### DISCUSSION TITLE
Any advice on how to get Mage Target data exported for usage in another Mage task?

### DISCUSSION THREAD

**Discussion by sheldonhull** [magefile/mage#368] [Q&A] [answered]
@natefinch I'm interested in writing a mage helper package that could generate Jetbrains and VSCode task schema so there is a GUI driven option to invoke tasks (at least ones without parameters). Additionally, I used this with other task frameworks to generate a help page of documentation for the tasks. I'm assuming I can solve that one just with go documentation tooling, but figured I'd check as well. 

I found this a nice experience for new contributors.

![2021-09-21-11 23 13-Code](https://user-images.githubusercontent.com/3526320/134209157-72b746a6-c0e1-4515-ab86-0f8b6738aa12.png "An example with VSCode using Task Explorer")
An example with VSCode using Task Explorer

Any pointers you could give me as I'm new to the source of mage, that could point me towards the discovered targets. 
If target discovery doesn't get exposed till the final compilation, then perhaps this should turn into a proposal for Mage?

Perhaps a new command `mage generate tasks --schema vscode` or something like this?

**Comment by natefinch** [comment, marked as answer]
I've been wanting to do this for forever, but never could get past the docs on integrating with VSCode. 

Probably the best way to do this is by adding a command to the compiled binary that we create from the magefiles to output the right info for VSCode or whatever, then you only compile and parse that one time.... basically the exact same thing we do for `mage -l`, except instead of the CLI output, we output whatever vscode wants.

**Reply by sheldonhull** [reply]
Ok, will fork this discussion into an issue if I get time to contribute towards that. Thanks for the clarity!

**Reply by sheldonhull** [reply]
@natefinch  we chatted in slack on this and you said it's still a challenge as no full metadata captured in struct right now. Just capturing that here so I don't forget. 

> Re: task tree, there isn't a tree per se. There's a list of valid targets you could get, but the tree is implicit, since it's just code calling functions via sync.Once
